lib/msf/base/sessions/ldap.rb

```diff
@@ -139,4 +146,32 @@ def _interact_stream
     raise EOFError if (console.stopped? == true)
   end
 
+  def on_registered
+    start_keep_alive_loop
+  end
+
+  # Start a background thread for regularly sending a no-op command to keep the connection alive
+  def start_keep_alive_loop
+    self.keep_alive_thread = framework.threads.spawn('LDAP-shell-keepalive', false) do
```

Comment:
Not a blocker: It might be handy to have the session ID into this thread name as metadata, and in the [WinRM implementation](https://github.com/rapid7/metasploit-framework/blob/5e2fab24efb826bc77362ceb0a38a38305d8d90d/lib/msf/base/sessions/winrm_command_shell.rb#L90)

Reply:
I've added it to the name.